packages/itinerary-body/src/AccessLegBody/index.js

```diff
@@ -0,0 +1,302 @@
+
+/**
+ * Component for access (e.g. walk/bike/etc.) leg in narrative itinerary. This
+ * particular component is used in the line-itin (i.e., trimet-mod-otp) version
+ * of the narrative itinerary.
+ */
+export default class AccessLegBody extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { expanded: false };
+  }
+
+  onStepsHeaderClick = () => {
+    const { expanded } = this.state;
+    this.setState({ expanded: !expanded });
+  };
+
+  onSummaryClick = () => {
+    const { leg, legIndex, setActiveLeg } = this.props;
+    setActiveLeg(legIndex, leg);
+  };
+
+  render() {
+    const {
+      config,
+      diagramVisible,
+      followsTransit,
+      leg,
+      LegIcon,
+      routingType,
+      setLegDiagram,
+      showElevationProfile,
+      timeOptions
+    } = this.props;
+    const { expanded } = this.state;
+
+    if (leg.mode === "CAR" && leg.hailedCar) {
+      return (
+        <TNCLeg
+          config={config}
+          leg={leg}
+          LegIcon={LegIcon}
+          onSummaryClick={this.onSummaryClick}
+          timeOptions={timeOptions}
+          followsTransit={followsTransit}
+        />
+      );
+    }
+
+    return (
+      <StyledLegs.LegBody>
+        <AccessLegSummary
+          config={config}
+          leg={leg}
+          LegIcon={LegIcon}
+          onSummaryClick={this.onSummaryClick}
+        />
+        <Styled.StepsHeader onClick={this.onStepsHeaderClick}>
+          {formatDuration(leg.duration)}
+          {leg.steps && (
+            <span>
+              {" "}
+              <StyledLegs.CaretToggle expanded />
+            </span>
+          )}
+        </Styled.StepsHeader>
+
+        {routingType === "ITINERARY" && (
+          <LegDiagramPreview
+            diagramVisible={diagramVisible}
+            leg={leg}
+            setLegDiagram={setLegDiagram}
+            showElevationProfile={showElevationProfile}
+          />
+        )}
+        <VelocityTransitionGroup
+          enter={{ animation: "slideDown" }}
+          leave={{ animation: "slideUp" }}
+        >
+          {expanded && <AccessLegSteps steps={leg.steps} />}
+        </VelocityTransitionGroup>
+      </StyledLegs.LegBody>
+    );
+  }
+}
+
+AccessLegBody.propTypes = {
+  config: configType.isRequired,
+  /**
+   * Should be either null or a legType. Indicates that a particular leg diagram
+   * has been selected and is active.
+   */
+  diagramVisible: legType,
+  followsTransit: PropTypes.bool,
+  leg: legType.isRequired,
+  LegIcon: PropTypes.elementType.isRequired,
+  legIndex: PropTypes.number.isRequired,
+  routingType: PropTypes.string.isRequired,
+  setActiveLeg: PropTypes.func.isRequired,
+  setLegDiagram: PropTypes.func.isRequired,
+  showElevationProfile: PropTypes.bool,
+  timeOptions: timeOptionsType.isRequired
+};
+
+AccessLegBody.defaultProps = {
+  diagramVisible: null,
+  followsTransit: false,
+  showElevationProfile: false
+};
+
+function TNCLeg({
```

Comment:
Should this class be in its own file? (same question for AccessLegSummary)

Reply:
In the past, we haven't been too strict about having one component per file, but maybe it's more readable to do so.